src/IO/ReadWriteBufferFromHTTP.h

```diff
@@ -276,6 +277,8 @@ namespace detail
                                 "0 < http_retry_initial_backoff_ms < settings.http_retry_max_backoff_ms (now 0 < {} < {})",
                                 settings.http_max_tries, settings.http_retry_initial_backoff_ms, settings.http_retry_max_backoff_ms);
 
+            http_header_entries.emplace_back(std::make_pair("User-Agent", fmt::format("ClickHouse/{}", VERSION_STRING)));
```

Comment:
What if User-Agent is already specified in the headers?

Reply:
Good point. The most efficient way to check this would be a linear traversal of the vector `O(n)`. Loading everything into a map would use memory and not improve efficiency.